### PR TITLE
Reactions

### PR DESCRIPTION
**Pull request by AlaaN-Smadi**

### CODE REVIEW

File: scripts/buildfire/components/reactions/reactions.js [hunk 1]
```diff
@@ -0,0 +1,1594 @@
...
+                return console.error("callback must be a function!");
+            }
+
+            buildfire.appData.get(this.TAG, (err, result) => {
+                if (err) return callback(err)
+
+                if (!result.data || !result.data.groups || !result.data.groups.length) {
+                    console.error('No reaction groups added yet');
+                    this.groups = [];
+                    return callback(null, []);
```
Comment (mas-iota) [inline scripts/buildfire/components/reactions/reactions.js:605]: callback err

File: scripts/buildfire/components/reactions/reactions.js [hunk 2]
```diff
@@ -0,0 +1,1594 @@
...
+
+            let group = null;
+            if (groupName) {
+                for (let i = 0; i < this.groups.length; i++) {
+                    if (this.groups[i].name.toLowerCase() === groupName.toLowerCase()) {
+                        group = this.groups[i];
+                        break;
+                    }
+                }
+                if (!group) {
```
Comment (mas-iota) [inline scripts/buildfire/components/reactions/reactions.js:637]: should return an err

File: scripts/buildfire/components/control/reactions/reactionGroupsPicker.js [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+                return this.updateContentState('empty');
+            }
+            let listContainer = this.dialogContainer.querySelector('#reactions-dialog-list-container');
+    
+            let list = '';
+            options.groups.forEach((group, idx) => {
+                let reactions = '';
+                group.reactions.forEach(reaction => {
+                    let src = buildfire.imageLib.resizeImage(reaction.selectedUrl, { size: "half_width", aspect: "1:1" })
+                    reactions += `<img style="width:24px; height:24px;" class="margin-right-five" src="${src}" />`;
```
Comment (mas-iota) [inline scripts/buildfire/components/control/reactions/reactionGroupsPicker.js:235]: half_width is too much

File: scripts/buildfire/components/reactions/reactions.js [hunk 4]
```diff
@@ -0,0 +1,1594 @@
...
+                    group = this.groups[0];
+                }
+            } else {
+                group = this.groups[0];
+            }
+
+            let allActiveReactions = group.reactions.filter(reaction => (reaction.isActive == true && reaction.selectedUrl && reaction.unSelectedUrl));
+            // resize reaction images 
+            allActiveReactions = allActiveReactions.map(reaction => {
+                let selectedUrl = buildfire.imageLib.resizeImage(reaction.selectedUrl, { size: "m", aspect: "1:1" }),
```
Comment (mas-iota) [inline scripts/buildfire/components/reactions/reactions.js:647]: "m" is too much

**Review by mas-iota — APPROVED** [review]

**Review by bkalaldeh — APPROVED** [review]

**Comment by ahabeb** [comment]
> "sx" in buildfire.imageLib.resizeImage ?

make sense, @AlaaN-Smadi can you update that?

**Review by bkalaldeh — APPROVED** [review]

**Review by bkalaldeh — APPROVED** [review]

**Pull request merged**